CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+    message (FATAL_ERROR "san option only works for clang/gcc")
+  endif ()
+  if (san STREQUAL "memory" AND NOT (is_clang AND
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7))
+    message (FATAL_ERROR "memory sanitizer only available for clang 7+")
+  endif ()
+endif ()
+
+# the remaining options are obscure and rarely used
+option (beast_no_unit_test_inline
+  "Prevents unit test definitions from being inserted into global table"
+  OFF)
+option (beast_force_debug
+  "Force BEAST_DEBUG regardless of DEBUG settings"
+  OFF)
+# NOTE - THIS OPTION CURRENTLY DOES NOT COMPILE :
+# TODO: fix or remove
+option (verify_nodeobject_keys
+  "This verifies that the hash of node objects matches the payload. \
+  This check is expensive - use with caution."
+  OFF)
+option (dump_leaks_on_exit
+  "Displays heap blocks and counted objects which were not disposed of\
+  during exit. Only implemented for windows builds."
+  ON)
+option (single_io_service_thread
+  "Restricts the number of threads calling io_service::run to one. \
+  This can be useful when debugging."
+  OFF)
+option (boost_show_deprecated
+  "Allow boost to fail on deprecated usage. Only useful if you're trying\
+  to find deprecated calls."
+  OFF)
+# beast_check_mem_leaks can't be an option() because we want to support an
+# "undefined" which means use default behavior
+# option (beast_check_mem_leaks
+#   "Force beast mem leak checking. Default is on for DEBUG builds. Only implemented on WIN32"
+if (WIN32)
+  option (beast_disable_autolink "Disables autolinking of system libraries on WIN32" OFF)
+else ()
+  set (beast_disable_autolink OFF CACHE BOOL "WIN32 only" FORCE)
+endif ()
+if (coverage)
+  message (STATUS "coverage build requested - forcing Debug build")
+  set (CMAKE_BUILD_TYPE Debug)
+endif ()
+
+if (is_linux AND NOT static)
+  set (BUILD_SHARED_LIBS ON)
+else ()
+  if (BUILD_SHARED_LIBS)
+    message (WARNING "forcing BUILD_SHARED_LIBS OFF for this platform/config")
+  endif ()
+  set (BUILD_SHARED_LIBS OFF)
+endif ()
+
+#[===================================================================[
+   setup project-wide compiler settings
+#]===================================================================]
+
+#[=========================================================[
+   TODO some/most of these common settings belong in a
+   toolchain file, especially the ABI-impacting ones
+#]=========================================================]
+add_library (common INTERFACE)
+add_library (Ripple::common ALIAS common)
+# add a single global dependency on this interface lib
+link_libraries (Ripple::common)
+set_target_properties (common
+  PROPERTIES INTERFACE_POSITION_INDEPENDENT_CODE ON)
+target_compile_features (common INTERFACE cxx_std_14)
+target_compile_definitions (common
+  INTERFACE
+    $<$<CONFIG:Debug>:_DEBUG>
+    $<$<AND:$<BOOL:profile>,$<NOT:$<BOOL:assert>>>:NDEBUG>
+    OPENSSL_NO_SSL2         # << WHAT DOES THIS DO ?
+    _SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS)
```

Comment:
I would be quite surprised if we need this. Is it left-over from before or did you verify this is actually needed?

Reply:
most of the flags here are legacy and have been carried down from previous build systems. I'm not sure what's the best way to go about deciding when to remove some of these.

Reply:
..and it turns out you need this if you build the (very old) protobuf library sources (which effectively only happens on windows right now...). SO, I've narrowed the scope of this define down to just the protobuf lib

Reply:
Always something. :)

Thanks for tightening the scope on it and making it more explicit to future readers. Without seeing it as a Windows-only-thing, I had no way of knowing what is cross-platform vs. what is Windows only.